### PR TITLE
v0.4.10 changelog update

### PR DESCRIPTION
**Pull request by geoffthemedio**
WIP, needs key changes and some formatting, and some review...

### CODE REVIEW

**Comment by Vezzra** [comment]
Excellent, thanks for picking up that task again.

Looks good at a first glance. Regarding key changes I want the input of the current active contributors. The one thing I definitely want in that section is the switch to Python 3, as this has been the primary focus of the 0.4.10 release.

File: ChangeLog.md [hunk 1]
```diff
@@ -4,21 +4,172 @@ Changelog
...
+    - Adjusted costs and build times
+    - Species piloting traits give -1 to +3 for all fighters except interceptors
+
+- Ships
+    - Ship production cost increases ("upkeep") also count ships on the production queue
+    - Reduced Shield part costs
+    - Colony base hulls may not mount engines or fuel parts
+    - Set Colony Base hull structure to 2
+    - Scaled up ship structure to avoid fractional numbers
+    - Increased default combat rounds from 3 to 4 and rebalanced ship structure levels accordingly
```
Comment (agrrr3) [inline ChangeLog.md]: Not sure if default bouts == 4  made it into 0.4.10 (yet)

File: ChangeLog.md [hunk 2]
```diff
@@ -4,21 +4,172 @@ Changelog
...
+    - Fighter refinement techs increase interceptor capacity and launch rate, not damage
+    - Techs increase strike fighter damage by +2, bomber damage by +3, and heavy bomber damage by +6
+    - Adjusted costs and build times
+    - Species piloting traits give -1 to +3 for all fighters except interceptors
+
+- Ships
+    - Ship production cost increases ("upkeep") also count ships on the production queue
+    - Reduced Shield part costs
+    - Colony base hulls may not mount engines or fuel parts
+    - Set Colony Base hull structure to 2
```
Comment (agrrr3) [inline ChangeLog.md]: base hull is more expensive but has three internal slots

Comment (geoffthemedio) [reply ChangeLog.md]: you want to modify a line or add a line?

File: ChangeLog.md (hunk 1 above)
Comment (geoffthemedio) [reply ChangeLog.md]: The change seemingly was done but then undone...

https://github.com/freeorion/freeorion/commits/release-v0.4.10/combat/CombatEvents.cpp

**Comment by geoffthemedio** [comment]
updated...

File: ChangeLog.md [hunk 3]
```diff
@@ -4,21 +4,176 @@ Changelog
...
+    - Fixed descriptions of concentration camp effects
+
+- New Stuff
+    - Added Arc Disruptor ship weapon part
+    - Added sitreps for gifting to allied empires
+
+- Reworked what weapons can target fighters, ships, and planets
+    - Mass drivers, Lasers, Plasmas, Death Rays, and Spinal antimatter cannon attack ships and planets (no fighters)
+    - Interceptors attack enemy fighters only
+    - Fighters attack ships and fighters
```
Comment (agrrr3) [inline ChangeLog.md]: Strike fighters (formerly known as Fighter fighters) attack ships and fighters

File: ChangeLog.md [hunk 4]
```diff
@@ -4,21 +4,176 @@ Changelog
...
+
+- Scripting
+    - Optimized some condition internal evaluation and scripted implementation
+    - Optimized evaluation of effectsgroup scopes
+    - Optimized some effect execution
+    - Reworked hull and fuel regeneration part effects
+
+- Python AI
+    - Scripted AIs to scrap Gateways to the Void that they capture
+    - Reprioritized fuel tach research
```
Comment (agrrr3) [inline ChangeLog.md]: Reprioritized fuel tech research

File: ChangeLog.md [hunk 5]
```diff
@@ -4,21 +4,176 @@ Changelog
...
+- The server stop auto-advancing turns after a player has won the game
+- Relabelled Resign button to Exit to Menu
+- Removed unused / redundant keymaps configuration file
+- Repeatedly clicking to close the game window will override the default autosave and exit faster
+
+#### Content and Balance
+
+- Stringtables
+    - Added a few names to lists used for ship names
+    - Translation updates: Spanish
```
Comment (agrrr3) [inline ChangeLog.md]: and French and Russian(?)

File: ChangeLog.md [hunk 6]
```diff
@@ -4,21 +4,176 @@ Changelog
...
 
 
-[v0.4.9+] - TBA
+[v0.4.10] - TBA
 ---------------
 
+### Key Changes
+
+- Python 3.5+ is now supported and required. Python 2 dependence is removed
+- Added a new ship weapon part, the Arc Disruptor, which is useful for countering many cheap "chaff" ships
```
Comment (agrrr3) [inline ChangeLog.md]: Added a new ship weapon tech line, the Arc Disruptor, which is a good option for bad pilots, against comsats, against ships without shields, and has powerful part upgrades.

File: ChangeLog.md [hunk 7]
```diff
@@ -4,21 +4,176 @@ Changelog
...
+- Reworked weapon and fighter combat targeting to limit what each can target
```
Comment (agrrr3) [inline ChangeLog.md]: Fighter stats are completely rebalanced and the Heavy Bomber fighter type was added

File: ChangeLog.md [hunk 8]
```diff
@@ -4,21 +4,176 @@ Changelog
...
+    - Exposed empire stat records
+
+- FreeOrion Content Script Interface (FOCS)
+    - Added UsedInDesignID ValueRef to access the ID of the ship design in which a part or hull is used
+    - Added arrived-from system property value ref for fleets
+    - Added ShipDesignCost ComplexValueRef
+    - Made some ValueRefs return default values rather than throwing errors when invalid properties are evaluated
+    - Added missing empire ID handling in several conditions
+    - Replaced OwnerTradeStockpile with EmpireStockpile (that refers to industry stockpiled)
+    - Added NoDefaultFuelEffect flag to hulls
```
Comment (agrrr3) [inline ChangeLog.md]: Added NoDefaultSpeedEffect, NoDefaultFuelEffect, NoDefaultStealthEffect, and NoDefaultStructureEffect flag to hulls. Non-monster hulls now use NoDefaultFuelEffect to provide fuel efficiency

File: ChangeLog.md (hunk 5 above)
Comment (geoffthemedio) [reply ChangeLog.md]: There haven't been any substantial updates to ru.txt merged yet. @o01eg @naiffuR @Cjkjvfnby any idea if / when https://github.com/freeorion/freeorion/pull/2793 will be ready?

**Comment by geoffthemedio** [comment]
Updated... hopefully accurately...

**Comment by Vezzra** [comment]
Thanks! Will merge this before I build RC1 on Thusday.

Anyone who wants to review/comment is welcome to do so before.

File: ChangeLog.md [hunk 9]
```diff
@@ -4,21 +4,198 @@ Changelog
...
+- Repeatedly clicking to close the game window will override the default autosave and exit faster
+
+#### Content and Balance
+
+- New Stuff
+    - Added Arc Disruptor ship weapon part
+    - Added sitreps for gifting to allied empires
+
+- Reworked what weapons can target fighters, ships, and planets
+    - Mass drivers, Lasers, Plasmas, Death Rays, and Spinal antimatter cannon attack ships and planets (no fighters)
```
Comment (Cjkjvfnby) [inline ChangeLog.md]: `cannon attack ships and planets (no fighters)`

What can be attacked and what not?   Looks like everything cannot be attacked by these weapons.

Github markdown allows tables.  May be we should create a table:  `[weapon, can attack ships, can attack planet can attack any type of fighter]`?

File: ChangeLog.md [hunk 10]
```diff
@@ -4,21 +4,198 @@ Changelog
...
+
+- Rebalanced fighters
+    - Interceptors launch 50% faster than other fighters
+    - Fighter refinement techs increase interceptor capacity and launch rate, not damage
+    - Techs increase strike fighter damage by +2, bomber damage by +3, and heavy bomber damage by +6
+    - Adjusted costs and build times
+    - Species piloting traits give -1 to +3 for all fighters except interceptors
+
+- Ships
+    - Ship production cost increases ("upkeep") also count ships on the production queue
```
Comment (Cjkjvfnby) [inline ChangeLog.md]: `production queue` is ambiguous.  Do you mean ships in the production queue that already get any production points or all ships?

File: ChangeLog.md [hunk 11]
```diff
@@ -4,21 +4,198 @@ Changelog
...
+    - Normal industry and research growth only if planet happiness is at least five
+    - Force energy structure industry and research growth only if happiness is at least five
+    - Planets can be attacked in combat with nonzero infrastructure, in addition to nonzero defense or shields, so that they will be regularly attacked every turn in combat and thus not regenerate meters every other turn
+    - New colonies start with 1, 2, or 4 happiness depending on environment level
+    - Outposts have default 1 infratructure so that they can be attacked every turn
+    - Reduced some population-based bonuses to resource output to mitigate snowballing empire development
+    - Rebalanced some flat bonuses from techs and Gas Giant Generators
+    - Rebalance empire-wide pop-based specials and increase their protection
+    - Added one refinement level to Nascent Artificial Intelligence and Adaptive Automation techs
+    - Rebalanced Exobots to be adequate on all planet types and very bad a research
```
Comment (Cjkjvfnby) [inline ChangeLog.md]: `bad a research` -> bad at research

**Review by Cjkjvfnby — DISMISSED** [review]

File: ChangeLog.md (hunk 9 above)
Comment (geoffthemedio) [reply ChangeLog.md]: I don't understand the question... those weapons attack ships and planets, not fighters. Maybe no vs not was unclear? I can remove the (no fighters) part...

Not keen on assuming this text is only to be read on GitHub and thus using its markup.

File: ChangeLog.md (hunk 10 above)
Comment (geoffthemedio) [reply ChangeLog.md]: added `that have received some production points`

File: ChangeLog.md (hunk 9 above)
Comment (Cjkjvfnby) [reply ChangeLog.md]: > I don't understand the question...
 
Sorry I misread cannon and cannot.   Let it be as is.

**Pull request merged**